example/conf.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Example project to showcase the extension's configuration."""
+from datetime import datetime
+
+############################################################
+### Project information
+############################################################
+
+# Product name
+project = "Example Project"
+author = "Example Project Authors"
+
+# The title you want to display for the documentation in the sidebar.
+# You might want to include a version number here.
+# To not display any title, set this option to an empty string.
+html_title = project + " Docs"
+
+# The default value uses the current year as the copyright year.
+#
+# For static works, it is common to provide the year of first publication.
+# Another option is to give the first year and the current year
+# for documentation that is often changed, e.g. 2022–2023 (note the en-dash).
+#
+# A way to check a GitHub repo's creation date is to obtain a classic GitHub
+# token with 'repo' permissions here: https://github.com/settings/tokens
+# Next, use 'curl' and 'jq' to extract the date from the GitHub API's output:
+#
+# curl -H 'Authorization: token <TOKEN>' \
+#   -H 'Accept: application/vnd.github.v3.raw' \
+#   https://api.github.com/repos/canonical/<REPO> | jq '.created_at'
+copyright = f"{datetime.today().year}, {author}"
+
+# Set the path to your own static assets here
+html_static_path = ["_static"]
+
+# (Some settings must be part of the html_context dictionary, while others
+#  are on root level. Don't move the settings.)
```

Comment:
Since we're processing these settings in the extension anyway - would it be easier/useful to put them all on the top-level here?
And maybe prefix them with `canonical.` to make it clear that these are our custom settings?

Reply:
Or wait - do we actually process them in the extension **before** we get here, so we don't have that option?